kcp/controllers/kyma_controller.go

```diff
@@ -0,0 +1,160 @@
+
+//+kubebuilder:rbac:groups=kyma-project.io,resources=kymas,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kyma-project.io,resources=kymas/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kyma-project.io,resources=kymas/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciliation loop starting for", "resource", req.NamespacedName.String())
+	// check if kyma resource exists
+	kymaCR := &kyma.Kyma{}
+	if err := r.Get(ctx, req.NamespacedName, kymaCR); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info(req.NamespacedName.String() + " got deleted! Reference in WatcherCR ConfigMap will be removed")
+			// TODO: Delete Kyma from the corresponding ConfigMaps - will be implemented in next iteration
+			return ctrl.Result{}, nil //nolint:wrapcheck
+		}
+		return ctrl.Result{}, err //nolint:wrapcheck
+	}
+
+	// Kyma resource was created or updated
+	// Get installed modules from Kyma CR
+	modules := getModulesList(kymaCR)
+	if len(modules) == 0 {
+		return ctrl.Result{}, errors.New("module list of KymaCR is empty")
+	}
+	for _, module := range modules {
+		watcherCR, err := r.getWatcherCR(ctx, module, kymaCR.Namespace)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if value, ok := watcherCR.Labels[DefaultOperatorWatcherCRLabel]; ok && strings.ToLower(value) == "true" {
+			watcherConfigMap, err := r.getWatcherCM(ctx, module, kymaCR.Namespace)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			err = r.updateConfigMap(ctx, watcherConfigMap, kymaCR)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			logger.Info("Corresponding ConfigMap of WatcherCR got updated.")
+			return ctrl.Result{}, nil
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *KymaReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	SchemeBuilder := &scheme.Builder{GroupVersion: kyma.GroupVersion}
+	SchemeBuilder.Register(&kyma.Kyma{})
+	// Create ControllerBuilder
+	controllerBuilder := ctrl.NewControllerManagedBy(mgr).
+		For(
+			&kyma.Kyma{},
+			builder.WithPredicates(predicate.GenerationChangedPredicate{}),
+		)
+
+	return controllerBuilder.Complete(r)
+}
+
+func getModulesList(kymaResource *kyma.Kyma) []kyma.Module {
+	modules := kymaResource.Spec.Modules
+	return modules
+}
+
+func (r *KymaReconciler) getWatcherCR(ctx context.Context,
+	module kyma.Module,
+	namespace string,
+) (*componentv1alpha1.Watcher, error) {
+	watcherCRName := fmt.Sprintf("%s-%s", module.Name, module.Channel)
+	watcherCR := &componentv1alpha1.Watcher{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: namespace, Name: watcherCRName}, watcherCR); err != nil {
```

Comment:
I think you should skip if watcher CR was not found

Reply:
Makes sense, will implement it in this way